morf-core/src/main/java/org/alfasoftware/morf/upgrade/UpgradePathFinder.java

```diff
@@ -109,13 +109,13 @@ public SchemaChangeSequence determinePath(Schema current, Schema target, Collect
 
     // We have changes to make. Apply them against the current schema to see whether they get us the right position
     Schema trialUpgradedSchema = schemaChangeSequence.applyToSchema(current);
-    if (!schemasMatch(target, trialUpgradedSchema, "target domain schema", "trial upgraded schema", exceptionRegexes)) {
+    if (!schemasMatch(target, trialUpgradedSchema, "target domain schema (e.g. schema from Java source)", "database or xml schema with upgrades applied", exceptionRegexes)) {
       throw new NoUpgradePathExistsException();
     }
 
     // Now reverse-apply those changes to see whether they get us back to where we started
     Schema reversal = schemaChangeSequence.applyInReverseToSchema(trialUpgradedSchema);
-    if (!schemasMatch(reversal, current, "upgraded schema reversal", "current schema", exceptionRegexes)) {
+    if (!schemasMatch(reversal, current, "upgraded database or xml schema with upgrades reversed", "current database or xml schema", exceptionRegexes)) {
       throw new IllegalStateException("Upgrade reversals are invalid");
     }

```

Comment:
Let's revert these changes and wait for the consensus outcome of recently revived discussions.